### PR TITLE
More contribution friendliness

### PR DESCRIPTION
**Pull request by Timendus**
@Estus-Dev:

I discovered I was already working on a contributing text locally :see_no_evil: Sorry I never committed that. I've taken some of the good parts and added them to your text. I've also added a little hashing script for the Windows people who probably don't have access to sha1sum(?).

Any suggestions or things I've missed?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "validate-schema": "node ./scripts/validate.js",
     "check-formatting": "prettier --check .",
     "format": "prettier --write .",
+    "hash": "node ./scripts/hash.js $1",
```
Comment (Estus-Dev) [inline package.json:12]: This needs to not have the `$1`.

NPM will pass through arguments as long as they're not `-` flags, and it does not do anything with the `$1` so far as I'm aware. I double checked in Windows and WSL Ubuntu as that's what I happened to be running when I saw this PR.

**Review by Estus-Dev — CHANGES_REQUESTED** [review]
Looks good to me, apart from the `$1` being invalid in `package.json`. I've been bitten by script runners like NPM enough that that's the very first thing I tested 

I made a commit to remove it with the following message:
```
    fix: Implicitly pass arguments to hash script

    `$1` does not work here, but NPM will pass positional args through to
    the subcommand.

    I went back and forth on whether to include `--` before the args, as
    it's a good habit to prevent NPM gobbling up non-positional flags, but
    at the end of the day I think it's just noise for new users.
```

But I don't want to bother with yet another PR, it's probably easier for you to fix.

Other than that, it's all things that I probably should have elaborated more on, but I just wanted to get something decent out there that covered the common footguns. Thanks for the additions.

I tested the script _(with the change)_ and the SHA-1 hash is consistent between `sha1sum` and `npm run hash`. Not that I expected it to be different but still good to make sure.

File: package.json (hunk 1 above)
Comment (Estus-Dev) [inline package.json]: This needs to not have the `$1`. `"hash": "node ./scripts/hash.js",` will do the trick.

NPM will pass through arguments as long as they're not `-` flags, and it does not do anything with the `$1` so far as I'm aware. I double checked in Windows and WSL Ubuntu as that's what I happened to be running when I saw this PR.

Comment (Timendus) [inline package.json]:
```suggestion
    "hash": "node ./scripts/hash.js",
```

**Comment by Timendus** [comment]
I think this is what you mean :) Thanks for the review!

**Comment by Estus-Dev** [comment]
Yep, that’s the one. Glad to help!

**Pull request merged**